core/primitives/src/upgrade_schedule.rs

```diff
@@ -0,0 +1,150 @@
+use chrono::{DateTime, NaiveDateTime, ParseResult, Utc};
+use near_primitives_core::types::ProtocolVersion;
+
+/// Defines the point in time after which validators are expected to vote on the new protocol version.
+pub(crate) struct ProtocolUpgradeVotingSchedule(pub &'static str);
```

Comment:
I'm surprised that this is a str.

how about we keep ti as DateTime inside and implement the FromStr trait? 
(then you don't need things like 'is_valid')

Reply:
FromStr doesn't work as a const expression.
FromStr actually gives us a simpler interface but needs non-const initialization, PTAL.